gammapy/maps/wcs.py

```diff
@@ -882,6 +882,38 @@ def __repr__(self):
         )
         return str_
 
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            # check overall shape and axes compatibility
+            if self.data_shape != other.data_shape:
+                return False
+
+            result = True
+            for axis, otheraxis in zip(self.axes, other.axes):
+                result &= axis == otheraxis
+            # check WCS consistency
+            return result and self.wcs.wcs.compare(other.wcs.wcs, tolerance=1e-6)
+        return NotImplemented
```

Comment:
Maybe put this at the start:
```
if not isinstance(other, self.__class__):
    return NotImplemented
```
and then have the rest of the method without indentation?

It's kind of like input validation, and I always find it easiest to read and maintain if that is at the top.

Reply:
OK